File: valiha-hotel-admin/src/domain/entities/reservation.ts
import { Breakfast } from "./breakfast";
import { Client } from "./client";
import { Payment } from "./payment";
import { Room } from "./room";
import { Shuttle } from "./shuttle";

export interface Reservation {
  id: string;
  checkIn: string;
  checkOut: string;
  state: string;
  parking: boolean;
  client: Client;
  payment: Payment;
  rooms: Room[];
  shuttles: Shuttle[];
  breakfasts: Breakfast[];
}

File: valihahotel/notifications/src/main/java/com/valiha/notification/domain/Reservation.java
package com.valiha.notification.domain;

import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor
@Data
public class Reservation {

  private String id;
  private String checkIn;
  private String checkOut;
  private String state;
  private boolean parking;
  private int pax;
  private Client client;
  private Payment payment;
  private List<Room> rooms;
  private List<Shuttle> shuttles;
  private List<Breakfast> breakfasts;
}
